rllib/core/models/specs/checker.py

```diff
@@ -232,27 +238,55 @@ def wrapper(self, input_data, **kwargs):
             if cache and not hasattr(self, "__checked_input_specs_cache__"):
                 self.__checked_input_specs_cache__ = {}
 
-            checked_data = input_data
-            if input_specs:
-                spec = getattr(self, input_specs, "___NOT_FOUND___")
-                if spec == "___NOT_FOUND___":
-                    raise ValueError(f"object {self} has no attribute {input_specs}.")
-                if spec is not None:
-                    spec = convert_to_canonical_format(spec)
-                    checked_data = _validate(
-                        cls_instance=self,
-                        method=func,
-                        data=input_data,
-                        spec=spec,
-                        filter=filter,
-                        tag="input",
+            func_successfully_executed = False
+
+            initial_exception = None
+
+            if only_check_on_retry:
+                # Attempt to run the function without spec checking
+                try:
+                    output_data = func(self, input_data, **kwargs)
+                    func_successfully_executed = True
+                except Exception as e:
+                    initial_exception = e
+                    logger.error(
```

Comment:
I think it's good practice to generally log an encountered error on the error level.
If not, the logging level becomes useless because you either raise the Error (and don't need to log) or catch it and only warn.